waku/node/waku_node.nim

```diff
@@ -194,12 +194,15 @@ proc connectToNodes*(
 
 ## Waku Metadata
 
-proc mountMetadata*(node: WakuNode, clusterId: uint32): Result[void, string] =
+proc mountMetadata*(
+    node: WakuNode, clusterId: uint32, shards: seq[uint16] = @[]
+): Result[void, string] =
   if not node.wakuMetadata.isNil():
     return err("Waku metadata already mounted, skipping")
 
   let metadata = WakuMetadata.new(clusterId, node.enr, node.topicSubscriptionQueue)
-
+  for shard in shards:
+    metadata.shards.incl(uint32(shard))
```

Comment:
Yes? I find it weird to add this to `waku_node.nim` when this PR is about adapting the Canary to it. I would expect that if this line would be missing, then we would have some bug open about it

Reply:
> Yes? I find it weird to add this to `waku_node.nim` when this PR is about adapting the Canary to it. I would expect that if this line would be missing, then we would have some bug open about it

I think that @DarshanBPatel 's suggestion sounds very reasonable and looks more explicit. Also, it seems correct too because we are already passing the `clusterId` as a parameter in `mountMetadata`.

Nevertheless, you and SP might have more context than me about that and there might be a more idiomatic way to achieve that, i.e. the way you suggested (which I didn't see while I was adding my first comment .)

Reply:
The less we add to `wakunode` the better IMO. In this case, you can have `WakuMetadata` add shard by subscribing to them.

Reply:
Thanks for the feedback @SionoiS, @gabrielmer, and @Ivansete-status. I'll update the implementation to add shard information via ENR.